http3/client.go

```diff
@@ -208,8 +208,9 @@ func (c *SingleDestinationRoundTripper) roundTrip(req *http.Request) (*http.Resp
 		}
 	}
 
+	resp := &http.Response{}
```

Comment:
I'm not sure if this is the best decision. In the previous Pr, `doRequest` used to call the code to read the body after the HTTP status code is received. However, since we're now calling `openRequestStream` I also created the `*http.Response` here and pass it into `c.hconn.openRequestStream()` and `c.doRequest` (which, in turn, impacts how `responseFromHeaders` works).